builtin/v9/migration/verifreg.go

```diff
@@ -39,38 +35,23 @@ func migrateVerifreg(ctx context.Context, adtStore adt8.Store, priorEpoch abi.Ch
 	nextAllocationID := verifreg9.AllocationId(1)
 	allocationsMapMap := make(map[address.Address]*adt9.Map)
 	var dealAllocationTuples []DealAllocationTuple
-	var proposal market8.DealProposal
-	if err = proposals.ForEach(&proposal, func(dealID int64) error {
-		// Nothing to do for unverified deals
-		if !proposal.VerifiedDeal {
-			return nil
-		}
-
-		pcid, err := proposal.Cid()
-		if err != nil {
-			return err
-		}
-
-		isPending, err := pendingProposals.Has(abi.CidKey(pcid))
-		if err != nil {
-			return xerrors.Errorf("failed to check pending: %w", err)
-		}
 
-		// Nothing to do for not-pending deals
-		if !isPending {
-			return nil
+	for _, dealID := range pendingDeals {
+		proposal, err := proposals.GetDealProposal(dealID)
+		if err != nil || proposal == nil {
```

Comment:
proposal == nil should not be reachable but I figured an explicit error is minutely nicer than a panic

Reply:
Agreed